sklearn/linear_model/tests/test_logistic.py

```diff
@@ -89,6 +90,40 @@ def test_error():
         assert_raise_message(ValueError, msg, LR(max_iter="test").fit, X, Y1)
 
 
+def test_logistic_cv_mock_scorer():
+
+    class MockScorer(object):
+        def __init__(self):
+            self.calls = 0
+            self.scores = [0.1, 0.4, 0.8, 0.5]
+
+        def __call__(self, model, X, y, sample_weight=None):
+            score = self.scores[self.calls % len(self.scores)]
+            self.calls += 1
+            return score
+
+    mock_scorer = MockScorer()
+    Cs = [1, 2, 3, 4]
+    cv = 2
+
+    lr = LogisticRegressionCV(Cs=Cs, scoring=mock_scorer, cv=cv)
+    lr.fit(X, Y1)
+
+    # Cs[2] has the highest score (0.8) from MockScorer
+    assert_equal(lr.C_[0], [Cs[2]])
+
+    # scorer called 8 times (cv*len(Cs))
+    assert_equal(mock_scorer.calls, cv*len(Cs))
+
+    # reset mock_scorer
+    mock_scorer.calls = 0
+    pred = lr.predict(X)
+    custom_score = assert_warns(ChangedBehaviorWarning, lr.score, X, pred)
+
+    assert_equal(custom_score, mock_scorer.scores[0])
+    assert_equal(mock_scorer.calls, 1)
```

Comment:
Should we test that by default there is no warning? Or is that overkill? Otherwise looks good.

Reply:
I have added `test_logistic_cv_score_does_not_warn_by_default` to test that there is no warning by default.